File: packages/native-bridge/plugins/pdf-tools/ios/Sources/PdfToolsPlugin.swift
import Foundation
import Capacitor

/**
 * Stub implementation of the PdfTools plugin for iOS.  Once
 * implemented this plugin would allow you to perform vector redaction
 * and metadata removal on PDF documents.  For now it simply
 * rejects any calls with an informative error message.
 */
@objc(PdfToolsPlugin)
public class PdfToolsPlugin: CAPPlugin {
    @objc public func sanitizePdf(_ call: CAPPluginCall) {
        call.reject("PdfTools plugin sanitizePdf() is not implemented on iOS yet")
    }
}

File: packages/native-bridge/plugins/pdf-tools/android/src/main/java/com/cleanshare/pdftools/PdfToolsPlugin.java
package com.cleanshare.pdftools;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * Stub implementation of the PdfTools plugin for Android.  The PDF
 * tools plugin will ultimately provide a native implementation of
 * vector redactions and metadata removal for large documents.  At
 * present this stub rejects any calls to sanitizePdf() with a
 * placeholder message.
 */
@CapacitorPlugin(name = "PdfTools")
public class PdfToolsPlugin extends Plugin {
    @PluginMethod
    public void sanitizePdf(PluginCall call) {
        call.reject("PdfTools plugin sanitizePdf() is not implemented on Android yet");
    }
}